chain/syncer.go

```diff
@@ -151,11 +152,20 @@ func NewSyncer(cst *hamt.CborIpldStore, c consensus.Protocol, s syncerChainReade
 // otherwise resolved over the network.  This method will timeout if blocks
 // are unavailable.  This method is all or nothing, it will error if any of the
 // blocks cannot be resolved.
-func (syncer *Syncer) getBlksMaybeFromNet(ctx context.Context, blkCids []cid.Cid) ([]*types.Block, error) {
+func (syncer *Syncer) getBlksMaybeFromNet(ctx context.Context, blkCids []cid.Cid) ([]types.TipSet, error) {
 	ctx, cancel := context.WithTimeout(ctx, blkWaitTime)
 	defer cancel()
 
-	return syncer.fetcher.GetBlocks(ctx, blkCids)
+	tss, err := syncer.fetcher.FetchTipSets(ctx, types.NewTipSetKey(blkCids...), 1)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(tss) > 1 {
```

Comment:
I doubt we'll forget to nuke this but maybe leave a "// TODO this will go away once improved fetching lands" to keep other people wandering through this code from worrying.